src/checkout-buttons/strategies/masterpass-button-strategy.ts

```diff
@@ -0,0 +1,149 @@
+            return super.initialize(options);
+        }
+
+        return this._store.dispatch(this._checkoutActionCreator.loadDefaultCheckout())
+        .then(stateCheckout => {
+                this._stateCheckout = stateCheckout;
+
+                return this._masterpassScriptLoader.load(this._getPaymentMethod().config.testMode)
+                    .then(masterpass => {
+                        this._masterpassClient = masterpass;
+                        this._createSignInButton(masterpassOptions);
+                    });
+        })
+        .then(() => super.initialize(options));
+    }
+
+    deinitialize(options: CheckoutButtonOptions): Promise<void> {
+        if (!this._isInitialized) {
+            return super.deinitialize(options);
+        }
+
+        this._stateCheckout = undefined;
+        if (this._signInButton && this._signInButton.parentNode) {
+            this._signInButton.removeEventListener('click', this._handleWalletButtonClick);
+            this._signInButton.parentNode.removeChild(this._signInButton);
+            this._signInButton = undefined;
+        }
+
+        return super.deinitialize(options);
+    }
+
+    private _createSignInButton(masterpassOptions: MasterpassButtonInitializeOptions): void {
+        const { container } = masterpassOptions;
+        const buttonContainer = document.querySelector(`#${container}`);
+
+        if (!buttonContainer) {
+            throw new Error('Need a container to place the button');
+        }
+
+        const button = document.createElement('input');
+
+        button.type = 'image';
+        button.src = 'https://static.masterpass.com/dyn/img/btn/global/mp_chk_btn_160x037px.svg';
+        buttonContainer.appendChild(button);
+        this._signInButton = button;
+        this._signInButton.addEventListener('click', this._handleWalletButtonClick);
+    }
+
+    private _createMasterpassPayload(): MasterpassCheckoutOptions {
+        const paymentMethod = this._getPaymentMethod();
+        const checkout = this._getCheckout();
+
+        return {
+            checkoutId: paymentMethod.initializationData.checkoutId,
+            allowedCardTypes: paymentMethod.initializationData.allowedCardTypes,
+            amount: checkout.cart.cartAmount.toString(),
+            currency: checkout.cart.currency.code,
+            cartId: checkout.cart.id,
+            suppressShippingAddress: true,
+        };
+    }
+
+    @bind
+    private _handleWalletButtonClick(): void  {
+        const payload = this._createMasterpassPayload();
+        this._getMasterpassClient().checkout(payload);
+    }
+
+    private _getMasterpassClient(): Masterpass {
+        if (!this._masterpassClient) {
+            throw new NotInitializedError(NotInitializedErrorType.CheckoutButtonNotInitialized);
+        }
+
+        return this._masterpassClient;
+    }
+
+    private _getPaymentMethod(): PaymentMethod {
+        if (!this._methodId) {
+            throw new InvalidArgumentError();
```

Comment:
In this case, you should throw `NotInitializedError` because first of all this method doesn't accept an argument so it's odd to throw `InvalidArgumentError`, secondly the reason why `methodId` could be undefined is because the strategy is not initialized.